src/lsp-client/LSPEditorWrapper.cpp

```diff
@@ -834,6 +844,36 @@ LSPEditorWrapper::_DoFileStatus(nlohmann::json& params)
 	SetFileStatus(state.c_str());
 }
 
+void
+LSPEditorWrapper::_DoDocumentSymbol(nlohmann::json& params)
+{
+	BMessage msg('symb');
+	auto vect = params.get<std::vector<DocumentSymbol>>();
+	_DoRecursiveDocumentSymbol(vect, msg);
+
+	if (fEditor)
+		fEditor->SetDocumentSymbols(&msg);
+
+}
+
+void
+LSPEditorWrapper::_DoRecursiveDocumentSymbol(std::vector<DocumentSymbol>& vect, BMessage& msg)
+{
+	for (DocumentSymbol sym: vect) {
+		BMessage symbol;
+		symbol.AddString("name", sym.name.c_str());
+		symbol.AddInt32("kind", (int32)sym.kind);
+		BMessage child;
+		if (sym.children.size() > 0) {
+			_DoRecursiveDocumentSymbol(sym.children, child);
+		}
+		symbol.AddMessage("children", &child);
+		symbol.AddInt32("be:line", sym.selectionRange.start.line + 1);
```

Comment:
with the "Rename" commits I changed this to start:character and start:line (to match their end:character and end:line counterparts in the LSP Range structure), while keeping retro compatibility for be:line only. We probably may want to align to this naming convention maybe.

Reply:
You mean I should add "start:line" besides "be:line" and "start:character" besides "lsp:character" ?

Reply:
Done